File: src/main/java/com/github/nicsilver/jumpertest/action/JumperState.java
package com.github.nicsilver.jumpertest.action;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

@State(
        
        name = "com.github.nicsilver.jumpertest.action.JumperState",
        storages = {@Storage("SdkSettingsPlugin.xml")}
)

public class JumperState implements PersistentStateComponent<JumperState>
{
    public ArrayList<Integer> jumpCaretActions = new ArrayList<>();
    
    public static JumperState getInstance()
    {
        return ApplicationManager.getApplication().getService(JumperState.class);
    }
    
    @Nullable
    @Override
    public JumperState getState()
    {
        return this;
    }
    
    @Override
    public void loadState(@NotNull JumperState state)
    {
        XmlSerializerUtil.copyBean(state, this);
    }
    
    
}

File: src/main/kotlin/com/github/nicsilver/jumpertest/action/JumperState.kt
package com.github.nicsilver.jumpertest.action

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage

@State(
    name = "com.github.nicsilver.jumpertest.action.JumperState",
    storages = [Storage("SdkSettingsPlugin.xml")]
)
class JumperState : PersistentStateComponent<JumperState> {
    var jumpCaretActions: MutableList<Int> = mutableListOf()

    override fun getState(): JumperState = this

    override fun loadState(state: JumperState) {
        jumpCaretActions = state.jumpCaretActions
    }

    companion object {
        fun getInstance(): JumperState =
            ApplicationManager.getApplication().getService(JumperState::class.java)
    }
}
